[angkyakdifp] iP

angkyakdifp opened the pull request:
## Duke Pro
>“Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)
DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. download it from [here](https://github.com/angkyakdifp/ip/releases/tag/A-Jar).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

angkyakdifp committed:
Level-0

angkyakdifp committed:
Level-1

angkyakdifp committed:
Level-2

angkyakdifp committed:
Level-3

angkyakdifp committed:
Level-3

angkyakdifp committed:
Level-4

angkyakdifp committed:
TextUiTesting

angkyakdifp committed:
Level-5

angkyakdifp committed:
Level-6

angkyakdifp committed:
Level-7

angkyakdifp committed:
Level-7

angkyakdifp committed:
Level-8

angkyakdifp committed:
A-Package

angkyakdifp committed:
Resolving Conflict

angkyakdifp committed:
Merge pull request #1 from angkyakdifp/branch-Level-7

Level-7

angkyakdifp committed:
Merge pull request #2 from angkyakdifp/add-gradle-support

Add gradle support

angkyakdifp committed:
A-Gradle

angkyakdifp committed:
A-JUnit

angkyakdifp committed:
A-JavaDoc

angkyakdifp committed:
Merge pull request #3 from angkyakdifp/branch-A-JavaDoc

A-JavaDoc

angkyakdifp committed:
A-CodingStandard

angkyakdifp committed:
Merge pull request #4 from angkyakdifp/branch-A-CodingStandard

A-CodingStandard

angkyakdifp committed:
Level-9

angkyakdifp committed:
Merge pull request #5 from angkyakdifp/branch-A-CodingStandard

Level-9

angkyakdifp committed:
Level-9

angkyakdifp committed:
Merge pull request #6 from angkyakdifp/branch-Level-9

Level-9

Ruizhi2001 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,64 @@
+ *
+ * @author Angky Akdi Frandy Putrakelana
+ */
 public class Duke {
```
Should your Duke class be in a package as well? 😬

Ruizhi2001 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,87 @@
+     * @throws DukeException If there are an invalid Input.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
```
Can your method name be a bit more descriptive? 🤔 
Perhaps you could include what you are trying to execute.

Ruizhi2001 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @param description A string describing the task.
+     */
+    public AddCommand(String description) {
```
I like that you used PascalCase for class naming 👍

Ruizhi2001 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+     * @param storage The Storage that used to store, read and write data.
+     * @throws DukeException If there are an invalid Input.
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
```
Can your method name be a bit more descriptive? 🤔

Ruizhi2001 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,53 @@
+     * @throws DukeException If there are an invalid Input.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
```
Can your method name be a bit more descriptive? 🤔

Ruizhi2001 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+     *
+     * @return the string representing the Task that will be stored.
+     */
+    public abstract String getInput();
```
Are you getting input with this method?😰

Ruizhi2001 commented on src/test/java/duke/task/EventTest.java:
```diff
@@ -0,0 +1,41 @@
+import static org.junit.jupiter.api.Assertions.fail;
+public class EventTest {
+    @Test
+    public void invalidTime_exceptionThrown(){
```
Good use of underscores in test method names using the following three part format 👍

Ruizhi2001 commented on src/test/java/duke/command/AddCommandTest.java:
```diff
@@ -0,0 +1,34 @@
+
+public class AddCommandTest {
+    @Test
+    public void unknownCommand_exceptionThrown(){
```
Good use of underscores in test method names using the following three part format 👍

Ruizhi2001 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,99 @@
+     *
+     * @return the size of the list.
+     */
+    public int length() {
```
Should this method be named using a verb? 🤔

Ruizhi2001 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,128 @@
+     * Print the welcome message.
+     *
+     */
+    public void showWelcome() {
```
Good that the name representing the methods in this class are verbs and written in camelCase.👍

Ruizhi2001 reviewed (commented):
LGTM. Just some minor fixes.

angkyakdifp committed:
A-CheckStyle

angkyakdifp committed:
Merge pull request #7 from angkyakdifp/branch-A-CheckStyle

A-CheckStyle

angkyakdifp committed:
A-Level-10

angkyakdifp committed:
Merge pull request #8 from angkyakdifp/A-Level-10

A-Level-10

angkyakdifp committed:
CheckingStyle

angkyakdifp committed:
Fixing javadoc

angkyakdifp committed:
Fixing Method description

angkyakdifp committed:
Merge pull request #9 from angkyakdifp/branch-Level-10

Branch level 10

angkyakdifp committed:
[200~Enhance code to adhere to SLAP (Single Level of Abstraction Principle)

angkyakdifp committed:
Enhance code to adhere to SLAP (Single Level of Abstraction Principle).\n\nIn this commit, I have refactored the Main class of the Duke GUI application to improve code readability and maintainability by adhering to the Single Level of Abstraction Principle (SLAP). The code structure has been improved by separating concerns and making each method adhere to a single level of abstraction.\n\nKey Changes:- Separated the code into methods with clear responsibilities, making the code more modular and comprehensible.\n- Created methods for setting up the stage, scroll pane, anchor pane, and dialog.\n- Improved code comments and documentation for better clarity.\n- Ensured that each method follows SLAP by having a single level of abstraction.\n\nThese changes result in a more organized and maintainable codebase, making it easier to work with and understand.

angkyakdifp committed:
Merge branch 'master' of https://github.com/angkyakdifp/ip

angkyakdifp committed:
Enhance code to adhere to SLAP (Single Level of Abstraction Principle).+\n+\n+In this commit, I have refactored the Main class of the Duke GUI application to improve code readability and maintainability by adhering to the Single Level of Abstraction Principle (SLAP). The code structure has been improved by separating concerns and making each method adhere to a single level of abstraction.\n\nKey Changes:- Separated the code into methods with clear responsibilities, making the code more modular and comprehensible.\n- Created methods for setting up the stage, scroll pane, anchor pane, and dialog.\n- Improved code comments and documentation for better clarity.\n- Ensured that each method follows SLAP by having a single level of abstraction.\n\nThese changes result in a more organized and maintainable codebase, making it easier to work with and understand.

angkyakdifp committed:
Enhance code to adhere to SLAP (Single Level of Abstraction Principle).+\n+\n+In this commit, I have refactored the Main class of the Duke GUI application to improve code readability and maintainability by adhering to the Single Level of Abstraction Principle (SLAP). The code structure has been improved by separating concerns and making each method adhere to a single level of abstraction.\n\nKey Changes:- Separated the code into methods with clear responsibilities, making the code more modular and comprehensible.\n- Created methods for setting up the stage, scroll pane, anchor pane, and dialog.

- Improved code comments and documentation for better clarity.

- Ensured that each method follows SLAP by having a single level of abstraction.

These changes result in a more organized and maintainable codebase, making it easier to work with and understand.

angkyakdifp committed:
Merge branch 'master' of https://github.com/angkyakdifp/ip

angkyakdifp committed:
Enhance code to adhere to SLAP (Single Level of Abstraction Principle).

In this commit, I have refactored the Main class of the Duke GUI application to improve code readability and maintainability by adhering to the Single Level of Abstraction Principle (SLAP). The code structure has been improved by separating concerns and making each method adhere to a single level of abstraction.

Key Changes:- Separated the code into methods with clear responsibilities, making the code more modular and comprehensible.

- Created methods for setting up the stage, scroll pane, anchor pane, and dialog.

- Improved code comments and documentation for better clarity.

- Ensured that each method follows SLAP by having a single level of abstraction.

These changes result in a more organized and maintainable codebase, making it easier to work with and understand.

angkyakdifp committed:
Merge branch 'master' of https://github.com/angkyakdifp/ip

angkyakdifp committed:
Merge branch 'master' of https://github.com/angkyakdifp/ip

angkyakdifp committed:
Enhance code to adhere to SLAP (Single Level of Abstraction Principle).

In this commit, I have refactored the Main class of the Duke GUI application to improve code readability and maintainability by adhering to the Single Level of Abstraction Principle (SLAP). The code structure has been improved by separating concerns and making each method adhere to a single level of abstraction.

Key Changes:- Separated the code into methods with clear responsibilities, making the code more modular and comprehensible.

- Created methods for setting up the stage, scroll pane, anchor pane, and dialog.

- Improved code comments and documentation for better clarity.

- Ensured that each method follows SLAP by having a single level of abstraction.

These changes result in a more organized and maintainable codebase, making it easier to work with and understand.

angkyakdifp committed:
Enhance code to adhere to SLAP (Single Level of Abstraction Principle).

In this commit, I have refactored the Main class of the Duke GUI application to improve code readability and maintainability by adhering to the Single Level of Abstraction Principle (SLAP). The code structure has been improved by separating concerns and making each method adhere to a single level of abstraction.

Key Changes:- Separated the code into methods with clear responsibilities, making the code more modular and comprehensible.

- Created methods for setting up the stage, scroll pane, anchor pane, and dialog.

- Improved code comments and documentation for better clarity.

- Ensured that each method follows SLAP by having a single level of abstraction.

These changes result in a more organized and maintainable codebase, making it easier to work with and understand.

angkyakdifp committed:
Merge branch 'master' of https://github.com/angkyakdifp/ip

angkyakdifp committed:
Add Assertions

angkyakdifp committed:
Merge pull request #11 from angkyakdifp/branch-A-Assertsions

Add Assertions

angkyakdifp committed:
Improve Code Quality

angkyakdifp committed:
Merge pull request #12 from angkyakdifp/branch-A-CodeQuality

Improve Code Quality

angkyakdifp committed:
Add FixedTask Class

angkyakdifp committed:
Merge pull request #13 from angkyakdifp/branch-B-FixedDuration

Add FixedTask Class

angkyakdifp committed:
Creating User Guide

angkyakdifp committed:
Fixing User Guide

angkyakdifp committed:
Update README.md
